pkg/controller/service/servicecontroller.go

```diff
@@ -741,6 +741,7 @@ func (s *ServiceController) lockedUpdateLoadBalancerHosts(service *api.Service,
 		return nil
 	}
 
+	hosts := hostsFromNodeList(nodes)
```

Comment:
hostsFromNodeList() is only ever called after building up a NodeList using getNodeConditionPredicate() which check in the node is Unschedulable...

So (line 635-637) in hostsFromNodeList to is now redundant:
`		if list.Items[ix].Spec.Unschedulable {
			continue
		}
`
And can be removed